multi_physics/QED/QED_table_generator/table_generator.cpp

```diff
@@ -198,71 +298,440 @@ void generate_quantum_sync_photem_table(
+        << "  - chi_max   : " << params.chi_max << "\n"
+        << "  - frac_min  : " << params.frac_min << "\n"
+        << "  - chi_size  : " << params.chi_size << "\n"
+        << "  - frac_size : " << params.frac_size << "\n";
+
+    const auto dndt_params =
+        px_qs::dndt_lookup_table_params<RealType>{
+            params.chi_min,
+            params.chi_max,
+            params.chi_size};
+
+    const auto pair_prod_params =
+        px_qs::photon_emission_lookup_table_params<RealType>{
+            params.chi_min,
+            params.chi_max,
+            params.frac_min,
+            params.chi_size,
+            params.frac_size};
+
+    generate_quantum_sync_dndt_table<RealType, Policy>(dndt_params, file_name_prefix);
+    generate_quantum_sync_photem_table<RealType, Policy>(pair_prod_params, file_name_prefix);
+
+    std::cout << " ____________________________" << std::endl;
+}
 
-int main()
+/**
+* Prints an help message
+*/
+void print_help_message()
 {
+    std::cout << " ***** QED table generator HELP *****" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_HELP_S <<
+        " : prints this help message" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_HELP_L <<
+        " : prints this help message" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_TABLE <<
+        " : sets table type. Must be either " << OPT_TABLE_BREIT_WHEELER <<
+        " or " << OPT_TABLE_QUANTUM_SYNC << "." << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_PRECISION <<
+        " : sets table precision. Must be one of the following: "
+        << OPT_PRECISION_DOUBLE << ", "
+        << OPT_PRECISION_SINGLE << ", "
+        << OPT_PRECISION_SINGLE_WITH_DOUBLE_COMP << ".\n"
+        << std::setw(MAX_CMD_SIZE+3) << "" << OPT_PRECISION_SINGLE_WITH_DOUBLE_COMP
+        << " means that calculations are carried out in double precision\n"
+        << std::setw(MAX_CMD_SIZE+3) << ""
+        << "even if the table is in single precision. " << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_CHI_MIN <<
+        " : sets minimum chi parameter (real number, optional*)" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_CHI_MAX <<
+        " : sets maximum chi parameter (real number, optional*)" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_FRAC_MIN <<
+        " : sets minimum frac parameter (real number, optional*)" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_CHI_SIZE <<
+        " : sets number of chi parameters (integer number, optional*)" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_FRAC_SIZE <<
+        " : sets number of frac parameters (integer number, optional*)" << std::endl;
+
+    std::cout << std::setw(MAX_CMD_SIZE) << CMD_FILENAME <<
+        " : sets output file name (string)" << std::endl;
+
+    std::cout << " * Optional parameters have default values." << std::endl;
```

Comment:
I guess we would want to see the default values printed out if you write that. :p

Reply:
Sure!

Reply:
Now the help message contains the default values of the optional parameters:
```
 * Optional parameters have default values: 
   Breit-Wheeler tables: 
       --chi_min : 0.02
       --chi_max : 1000
       --frac_min : (unused)
       --chi_size : 256
       --frac_size : 256
   Quantum Synchrotron tables: 
       --chi_min : 0.001
       --chi_max : 1000
       --frac_min : 1e-12
       --chi_size : 256
       --frac_size : 256

```